File: python/7 kyu/348.py
# https://www.codewars.com/kata/580a4001d6df740d61000301/train/python

# Passed

def complete_series(seq): 
    val_to_count_dict = dict()
    for val in seq:
        if val not in val_to_count_dict:
            val_to_count_dict[val] = 0
            
        val_to_count_dict[val] += 1
        if val_to_count_dict[val] > 1:
            result = [0]
            break
    else:
        result = list(range(0, max(seq) + 1))
            
    return result

output = complete_series([1, 4, 6])
print(output)

File: nodejs/6 kyu/348.js
// https://www.codewars.com/kata/5515395b9cd40b2c3e00116c/train/javascript

// Passed

function regression_line(x, y) {
  let sumArr = (arr) => arr.reduce((prev, curr) => prev + curr, 0);
  let xiSum = sumArr(x);
  let yiSum = sumArr(y);

  let xiSquaredSum = sumArr(x.map((num) => num ** 2));

  let xiYiSum = sumArr(x.map((num, idx) => x[idx] * y[idx]));

  let a =
    (xiSquaredSum * yiSum - xiSum * xiYiSum) /
    (x.length * xiSquaredSum - xiSum ** 2);
  let b =
    (x.length * xiYiSum - xiSum * yiSum) /
    (x.length * xiSquaredSum - xiSum ** 2);
  let roundToFourDecimals = (num) => Math.round(num * 10 ** 4) / 10 ** 4;
  let result = [roundToFourDecimals(a), roundToFourDecimals(b)];

  return result;
}

const output = regression_line(
  [25, 30, 35, 40, 45, 50],
  [78, 70, 65, 58, 48, 42],
);
console.log(output);